### PR TITLE
[TEST] Consolidate GUI tests and expand coverage to edge cases

### PR DESCRIPTION
**Pull request by RainRat**
This PR improves the test suite's maintainability and reliability by consolidating fragmented GUI tests and filling significant coverage gaps.

**Type:** Reorg / New Coverage / Bug Fix

**What:**
- Unified 8 separate GUI-related test files into `tests/test_gui.py`.
- Developed a robust, centralized `mock_gui_deps` fixture for Tkinter mocking.
- Added coverage for:
    - Invalid regex handling in search (catches `tk.TclError`).
    - Zero-width regex matches (prevents infinite loops).
    - Hierarchical/threaded message insertion in the Treeview.
    - App exit logic.
    - Defensive sorting fallback.
- Moved Treeview item retrieval inside the `try` block in `sort_column` to prevent crashes on inconsistent UI state.

**Why:**
- Fragmentation of GUI tests made it difficult to maintain and discover existing tests.
- Several critical paths in the GUI (threading, regex edge cases, error fallbacks) were previously untested.
- The `sort_column` fix prevents non-deterministic crashes when sorting large lists or during state transitions.

---
*PR created automatically by Jules for task [5787021956518206119](https://jules.google.com/task/5787021956518206119) started by @RainRat*

### CODE REVIEW

File: pyqwk/gui.py [hunk 1]
```diff
@@ -507,14 +507,13 @@ def sort_column(self, col: str, reverse: bool) -> None:
...
-            else (self.message_list.item(k, "text"), k)
-            for k in self.message_list.get_children("")
-        ]
-
         try:
+            l = [
+                (self.message_list.set(k, col), k)
+                if col != "#0"
+                else (self.message_list.item(k, "text"), k)
+                for k in self.message_list.get_children("")
```
[the remark this answers is not in the document]
Comment (RainRat) [reply pyqwk/gui.py:516]: @jules Look into this
